packages/front-end/components/Radix/Button.tsx

```diff
@@ -0,0 +1,99 @@
+import { Button as RadixButton, ButtonProps, Text } from "@radix-ui/themes";
+import { CSSProperties, ReactNode } from "react";
+import { Responsive } from "@radix-ui/themes/dist/cjs/props";
+import Link from "next/link";
+import ConditionalWrapper from "@/components/ConditionalWrapper";
+
+type Overwrite<T, NewT> = Omit<T, keyof NewT> & NewT;
+
+export type Color = "primary" | "danger";
+export type Variant = "solid" | "soft" | "outline" | "ghost";
+export type Size = "xs" | "sm" | "md";
+
+export type Props = Overwrite<
+  ButtonProps,
+  {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    onClick?: (e: any) => void;
+    href?: string;
+    color?: Color | ButtonProps["color"];
```

Comment:
Only allow the colors from our design system, not all of the Radix colors.  That might mean we can't easily migrate as many buttons to the new component yet until they are redesigned, but that's fine.